files/justfiles/toggles.just

```diff
@@ -177,22 +177,84 @@ toggle-xwayland ACTION="prompt":
-        for file in "${BASH_ENV_FILES[@]}"; do
-            run0 chattr -i "$file"
+    read -r ACCEPT
+    if [ "$ACCEPT" != "YES I UNDERSTAND" ]; then
+        exit
+    fi
+    echo "Do you want this change to apply to ""${b}""all""${n}"" users (""${user_list[@]}"")? [y/N]"
+    echo "Otherwise, it will only apply to the currently logged in user ($SUDO_USER)."
+    read -r REPLY
+    if [[ ! "$REPLY" == [yY]* ]]; then
+    	user_list=("$SUDO_USER")
+    fi
+    for user in "${user_list[@]}"; do
+    	echo "Working user:$user"
+    	user_home=$(getent passwd "$user" | awk -F':' '{ print $6}')
+    	BASH_ENV_FILES=("/etc/profile") #used because we know it exists, not that it needs to be immutable
+    	if [ "$change" == "locked" ]; then
+    		if [ -d "$user_home/.bashrc.d/" ]; then	
+    			echo "$user_home/.bashrc.d/ found" # performative, not needed
+        	else
+                mkdir -p "$user_home/.bashrc.d/" #will create folder if necessary
+                chown -R "$user:" "$user_home/.bashrc.d/"
+        	fi
+    		chattr -R +i "$user_home/.bashrc.d/"
+    		if [ -d "$user_home/.config/environment.d/" ]; then
+                echo "$user_home/.config/environment.d/ found" # performative, not needed
+        	else
+        		mkdir -p "$user_home/.config/environment.d/" #will create folder if necessary
+            	chown -R "$user:" "$user_home/.config/environment.d/"
+        	fi
+            chattr -R +i "$user_home/.config/environment.d/"
+     	else
+     		if [ -d "$user_home/.bashrc.d/" ]; then	
+    			chattr -R -i "$user_home/.bashrc.d/"
+     		fi
+    		if [ -d "$user_home/.config/environment.d/" ]; then
+            		chattr -R -i "$user_home/.config/environment.d/"
+     		fi
+        fi
+    	BASH_FILES=("$user_home/.bashrc" "$user_home/.bash_profile" "$user_home/.config/bash-completion" "$user_home/.profile" "$user_home/.bash_logout" "$user_home/.bash_login")
+    	for file in "${BASH_FILES[@]}"; do
+            if [ -f "$file" ]; then
+                echo "$file found"
+            else
+                echo "No $file file found"
+                if [ "$change" == "locked" ]; then
+                    echo "Creating empty $file"
+                    install -D -o "$user" -g "$user" -m 600 "/dev/null" "$file" #will create folder if necessary
+                fi
+            fi
+    	    BASH_ENV_FILES+=("$file")
         done
-      else
-        echo "Bash environment '(${BASH_ENV_FILES[@]})' is unlocked. Locking it."
-        echo "
+    	if [ "$change" == "unlocked" ]; then
+            echo "Bash environment '(" "${BASH_ENV_FILES[@]}" ")' is locked down. Unlocking it."
+            for file in "${BASH_ENV_FILES[@]}"; do
+                chattr -i "$file"
+            done
+        else
+            echo "Bash environment '(" "${BASH_ENV_FILES[@]}" ")' is unlocked. Locking it."
+            for file in "${BASH_ENV_FILES[@]}"; do
+                chattr -i "$file"
```

Comment:
May be add a comment for future maintainers why the files are unlocked before locking them again?